### PR TITLE
setwarp owner uuid in yml file

### PR DESCRIPTION
**Pull request by FlashLight34**
To know who make the warp, so we can remove all warps of banned player and more :P

### CODE REVIEW

File: Essentials/src/com/earth2me/essentials/commands/Commandsetwarp.java [hunk 1]
```diff
@@ -36,7 +36,7 @@ public void run(final Server server, final User user, final String commandLabel,
         }
 
         if (warpLoc == null || user.isAuthorized("essentials.warp.overwrite." + StringUtil.safeString(args[0]))) {
-            warps.setWarp(args[0], loc);
+            warps.setWarp(user,args[0], loc);
```
Comment (mdcfe) [inline Essentials/src/com/earth2me/essentials/commands/Commandsetwarp.java]: Needs a space after `user,` and before `args[0]`.

**Review by mdcfe — COMMENTED** [review]
This looks good aside from a minor style issue.

Are you planning to implement `getLastOwner` in this PR?

**Comment by FlashLight34** [comment]
that it ;)

**Comment by FlashLight34** [comment]
its not ok?

File: Essentials/src/com/earth2me/essentials/Warps.java [hunk 2]
```diff
@@ -57,6 +57,10 @@ public Location getWarp(String warp) throws WarpNotFoundException, InvalidWorldE
 
     @Override
     public void setWarp(String name, Location loc) throws Exception {
+        setWarp(null,name,loc);
+    }
+    @Override
```
Comment (mdcfe) [inline Essentials/src/com/earth2me/essentials/Warps.java]: Should be a line space above this line.

File: Essentials/src/com/earth2me/essentials/Warps.java [hunk 3]
```diff
@@ -69,13 +73,26 @@ public void setWarp(String name, Location loc) throws Exception {
...
         conf.setProperty("name", name);
+        if (user != null) conf.setProperty("lastowner", user.getBase().getUniqueId().toString());
         try {
             conf.saveWithError();
         } catch (IOException ex) {
             throw new IOException(tl("invalidWarpName"));
         }
     }
-
+    @Override
```
Comment (mdcfe) [inline Essentials/src/com/earth2me/essentials/Warps.java]: Should also be a line space above this one too.

File: Essentials/src/com/earth2me/essentials/Warps.java [hunk 4]
```diff
@@ -69,13 +73,26 @@ public void setWarp(String name, Location loc) throws Exception {
...
+            throw new WarpNotFoundException();
+        }
+        UUID uuid = null;
+        try {
+            uuid = UUID.fromString(conf.getString("lastowner"));
+        }
+        catch (Exception ex) {}
+        return uuid;
+    }
     @Override
```
Comment (mdcfe) [inline Essentials/src/com/earth2me/essentials/Warps.java]: Line space needed above this line as well.

**Review by mdcfe — COMMENTED** [review]
Once these spacing issues are removed, this should be okay to merge.

**Comment by FlashLight34** [comment]
oh now its ok sorry about that :P

**Review by mdcfe — DISMISSED** [review]

File: Essentials/src/com/earth2me/essentials/Warps.java [hunk 5]
```diff
@@ -69,13 +72,28 @@ public void setWarp(String name, Location loc) throws Exception {
...
         try {
             conf.saveWithError();
         } catch (IOException ex) {
             throw new IOException(tl("invalidWarpName"));
         }
     }
-
+    
+    @Override
+    public UUID getLastOwner(String warp) throws WarpNotFoundException, InvalidWorldException {
```
Comment (drtshock) [inline Essentials/src/com/earth2me/essentials/Warps.java]: Why do we need to throw an invalid world exception?

Comment (FlashLight34) [reply Essentials/src/com/earth2me/essentials/Warps.java]: hehe i dont see this sorry ;)

**Review by mdcfe — APPROVED** [review]

**Pull request merged**